src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -106,11 +106,21 @@ StartRemoteTransactionBegin(struct MultiConnection *connection)
 		appendStringInfoString(beginAndSetDistributedTransactionId, activeSetStmts->data);
 	}
 
+	char *assignDistributedTransactionIdCommand = AssignDistributedTransactionIdCommand();
+
 	/* add SELECT assign_distributed_transaction_id ... */
 	appendStringInfoString(beginAndSetDistributedTransactionId,
-						   AssignDistributedTransactionIdCommand());
+						   assignDistributedTransactionIdCommand);
+
+	pfree(assignDistributedTransactionIdCommand);
```

Comment:
wondering what's a good place to add a comment on why we aggressively pfree here (either in function descriptions or above the relevant lines?)

Reply:
Added a comment to top of the file because it's not only that function but also some other doing aggressive pfree'ing.